keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+to isolate these 2 traffic types. To achieve this I require 2 separate network
+interfaces inside my Pod. The underlying implementation (outside the Pod) has to
+ensure isolation on either Layer-2 or Layer-3.
+
+<p align="center">
+  <img src="mn-story-1.png?raw=true" alt="multi-network story 1 network L2 isolation"/>
+</p>
+
+#### Story #2
+As a Cloud Native Network Function (CNF) vendor I require a HW-based interface
+(e.g. SRIOV VF) to be provisioned to my workload Pod. I need to leverage that HW
+for performance purposes (high bandwidth, low latency), that my user-space
+application (e.g. DPDK-based) can use. The VF will not use the standard netdev
+kernel module. The Pod’s scheduling to the nodes should be based on hardware
+availability (e.g. devicePlugin or some other way). This interface might not
+support some Kubernetes functionality like e.g. Services, I am willing to give
+that up for performance.
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster operator I wish to isolate workloads based on
+namespaces and network access by connecting to only a non- Cluster Default PodNetwork.
+Those workloads should have the same level of Kubernetes functionality:
+Services, NetworkPolicies, access to Kubernetes API.
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As Cluster operator I wish to manage my cluster’s bandwidth usage on a per-pod
+basis, simultaneously preserving all other Pod networking configuration the same
+across those Pods.
```

Comment:
The more I think about this, the less it seems to fit with this KEP.

Or at least, it doesn't fit _as a PodNetwork_. This seems more like a PodNetworkAttachment that points to the `default` PodNetwork but also includes a ParametersRef pointing to some well-known parameter type containing bandwidth information...

(And maybe that's actually what you meant? I guess it's not clear from the story itself...)

Reply:
@danwinship you have references to `PodNetworkAttachment` as in different places, and if for that you mean the representation on a Pod of a network interface https://docs.kernel.org/networking/netdevices.html, then this is the part I agree 100% we need to solve first.

And this is what actually DRA is solving https://github.com/kubernetes/enhancements/issues/3063 , it seems that the API is still revisiting and based on my current reading it can accomodate perfectly the use cases of `PodNetworkAttachment`

Once we have this capabilities we can use them to build the other high level abstractions required in this KEP